florist/api/db/entities.py

```diff
@@ -67,10 +75,127 @@ class Job(BaseModel):
     redis_port: Optional[Annotated[str, Field(...)]]
     clients_info: Optional[Annotated[List[ClientInfo], Field(...)]]
 
+    @classmethod
+    async def find_by_id(cls, job_id: str, database: AsyncIOMotorDatabase) -> Optional["Job"]:
+        """
+        Find a job in the database by its id.
+
+        :param job_id: (str) the job's id.
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        :return: (Optional[Job]) An instance of the job record with the given ID, or `None` if it can't be found.
+        """
+        job_collection = database[JOB_COLLECTION_NAME]
+        result = await job_collection.find_one({"_id": job_id})
+        if result is None:
+            return result
+        return Job(**result)
+
+    @classmethod
+    async def find_by_status(cls, status: JobStatus, limit: int, database: AsyncIOMotorDatabase) -> List["Job"]:
+        """
+        Return all jobs with the given status.
+
+        :param status: (JobStatus) The status of the jobs to be returned.
+        :param limit: (int) the limit amount of records that should be returned.
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        :return: (List[Job]) The list of jobs with the given status int the database.
+        """
+        status = jsonable_encoder(status)
+
+        job_collection = database[JOB_COLLECTION_NAME]
+        result = await job_collection.find({"status": status}).to_list(limit)
+        assert isinstance(result, list)
+        return [Job(**r) for r in result]
+
+    async def create(self, database: AsyncIOMotorDatabase) -> str:
+        """
+        Save this instance under a new record in the database.
+
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        :return: (str) the new job record's id.
+        """
+        json_job = jsonable_encoder(self)
+        result = await database[JOB_COLLECTION_NAME].insert_one(json_job)
+        assert isinstance(result.inserted_id, str)
+        return result.inserted_id
+
+    async def set_uuids(self, server_uuid: str, client_uuids: List[str], database: AsyncIOMotorDatabase) -> None:
+        """
+        Save the server and clients' UUIDs in the database under the current job's id.
+
+        :param server_uuid: [str] the server_uuid to be saved in the database.
+        :param client_uuids: List[str] the list of client_uuids to be saved in the database.
+        :param database: (motor.motor_asyncio.AsyncIOMotorDatabase) The database where the job collection is stored.
+        """
+        assert self.clients_info is not None and len(self.clients_info) == len(client_uuids), (
+            "self.clients_info and client_uuids must have the same length "
+            f"({'None' if self.clients_info is None else len(self.clients_info)}<>{len(client_uuids)})."
```

Comment:
nit: maybe using != instead of <> will be familiar for most python programmers. Also maybe its just me, but this is a bit hard to read